modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/bulk/TestBulkProcessor.java

```diff
@@ -138,6 +145,8 @@ public void testInvalidQuery() throws InterruptedException {
         assertEquals(COMPLETED, status.getState());
         assertEquals(0, status.getTotal());
         assertTrue(status.hasError());
+        int expectedStatus = coreFeature.getStorageConfiguration().isDBS() ? SC_NOT_FOUND : SC_INTERNAL_SERVER_ERROR;
```

Comment:
Why do we have a different behavior if DBS? Can you please add a comment?

Reply:
Yes, The tests are down, On Mongo this is related to the fact, that for SBS the http response is not `500` but `404`

```
[2021-03-18T17:55:12.373Z] [ERROR] testInvalidQuery(org.nuxeo.ecm.core.bulk.TestBulkProcessor)  Time elapsed: 0.425 s  <<< FAILURE!
[2021-03-18T17:55:12.373Z] java.lang.AssertionError: expected:<500> but was:<404>
```
More on https://jenkins.platform.dev.nuxeo.com/blue/organizations/jenkins/nuxeo%2Fnuxeo/detail/PR-4606/9/pipeline/109

Reply:
I think the point is to understand why the server itself returns different error codes, and see if that's a bug (to fix) or something we want (to document). In general that's not something we want, we prefer identical behavior, including error behavior, from backends.

Reply:
yes, agree on that. But I told myself if we find the issue and we align it to the same error code. It can break changes. But I'll take a look.

Reply:
I did a quick thing to use the `DocumentNotFoundException` too for `VCS` more on [Here](https://github.com/nuxeo/nuxeo/pull/4606#discussion_r598898734).